src/python/pants/build_graph/address.py

```diff
@@ -318,6 +318,12 @@ def is_default_target(self) -> bool:
     def is_parametrized(self) -> bool:
         return bool(self.parameters)
 
+    def is_parametrized_subset_of(self, other: Address) -> bool:
+        """True if this Address is == to the given Address, but with a subset of its parameters."""
+        if not self._equal_without_parameters(other):
+            return False
+        return self.parameters.items() <= other.parameters.items()
```

Comment:
> How would this work with configurations?

It should work fine with configurations, because it's based on field values. Configuration will be applied before the target(s) are actually created, and so the targets will have real field values, rather than indirection through the configuration.

> Also we talked last week about how alias expansion should be handled. I'm still a bit confused on that - I think it might work because `graph.py` normalizes things?

Not sure what you mean here.

Reply:
> Not sure what you mean here.

Making sure that this works properly - the alias is not a part of the identity.

```python
python_source(
   name="utils",
   source="utils.py",
   resolve=parametrize(a_alias="a", b_alias="b"),
)

python_source(
   name="app",
   source="app.py",
  resolve=paremetrize("a", "b"),
  dependencies=[":utils"],
)
```

Reply:
Yea, should work fine. The address starts out by needing to be a subset. If address1 specifies some-but-not-all of address2's parameters (verbatim, using aliases, or no), the remaining parameters are checked by field value. In the case you've given, `:utils` is a subset because it specifies no parameters at all: the remaining parameters ("resolve"), must then have equal field values.

Will cover it in `get_subset` tests though.